src/plugins/data_view_editor/public/components/data_view_flyout_content_container.tsx

```diff
@@ -32,6 +28,18 @@ const DataViewFlyoutContentContainer = ({
     services: { dataViews, notifications, http },
   } = useKibana<DataViewEditorContext>();
 
+  const dataViewEditorService = useRef(
+    new DataViewEditorService({
```

Comment:
I think that there is also a problem here that this `new DataViewEditorService({` is always called and created on each render. so all  the side effects and subscriptions inside the constructor repeat on each render even though they are not used

Reply:
https://blog.thoughtspile.tech/2021/11/30/lazy-useref/